xarray/core/computation.py

```diff
@@ -1306,3 +1307,86 @@ def where(cond, x, y):
         dataset_join="exact",
         dask="allowed",
     )
+
+
+def _calc_idxminmax(
+    *,
+    array,
+    func: Callable,
+    dim: Optional[Hashable],
+    skipna: Optional[bool],
+    promote: Optional[bool],
+    keep_attrs: Optional[bool],
+    **kwargs: Any,
+):
+    """Apply common operations for idxmin and idxmax."""
+    # This function doesn't make sense for scalars so don't try
+    if not array.ndim:
+        ValueError("This function does not apply for scalars")
+
+    if dim is not None:
+        pass  # Use the dim if available
+    elif array.ndim == 1:
+        # it is okay to guess the dim if there is only 1
+        dim = array.dims[0]
+    else:
+        # The dim is not specified and ambiguous.  Don't guess.
+        raise ValueError("Must supply 'dim' argument for multidimensional arrays")
+
+    if dim in array.coords:
```

Comment:
Let's also add a check that `dim` is in `array.dims`. Otherwise some other weird error could be triggered.

Reply:
Done